gk/main.c

```diff
@@ -96,6 +487,13 @@ run_gk(struct gk_config *gk_conf)
 	if (!gk_conf)
 		return -1;
 
+	gk_conf->num_lcores = gk_conf->lcore_end_id - gk_conf->lcore_start_id + 1;
+	gk_conf->instances = (struct gk_instance *)rte_calloc(NULL,
+		gk_conf->num_lcores, sizeof(struct gk_instance), 0);
```

Comment:
Maybe I'm missing something, but isn't that true for many errors? When an error occurs, the program will soon exit, but we still do some cleanup. Is there a difference here?